server/rest/src/main/java/org/infinispan/rest/resources/ContainerResource.java

```diff
@@ -434,7 +437,10 @@ private CompletionStage<RestResponse> listenConfig(RestRequest request) {
    }
 
    private CompletionStage<RestResponse> listenLifecycle(RestRequest request) {
-      return streamConfigurationAndEvents(request, true);
+      var requestedCategories = request.getParameter("category");
+      return requestedCategories == null || requestedCategories.isEmpty() ?
+            streamConfigurationAndEvents(request, true) :
```

Comment:
Potential future work (not this PR), could be to make it so that configuration changes are logged via the EventLog. That way we can have a single listener for all SSE and it would be possible for the Operator to just call `/v2/container?action=listen&category=cluster,lifecycle,config` in order to receive all required events \\cc @tristantarrant

Reply:
I'm not sure about the configuration events via the event log as you end up duplicating the cache configuration in both `CONFIG` and events cache. 

I would like to have a clearer definition of `/v2/container?action=listen`. In the docs is mentions that it sends `lifecycle` events only but in the code and in the test suite, it is sending the configuration events too with no way to disable them.

Reply:
> I'm not sure about the configuration events via the event log as you end up duplicating the cache configuration in both `CONFIG` and events cache.

Hmm yeah, that's not ideal. I guess the configurations aren't too large though, so maybe not too much of a concern?

One option would be to fake it and support a "config" category that only applies to the ContainerResource SSE listener and not `EventLoggerCategory`, but I don't really like this as it means that we have a discrepancy between Embedded and Server. I think we'll forget why we did it this way in a few years as well.


> I would like to have a clearer definition of `/v2/container?action=listen`. In the docs is mentions that it sends `lifecycle` events only but in the code and in the test suite, it is sending the configuration events too with no way to disable them.

The endpoint was originally created for the Operator with the idea of allowing it to just consume a single SSE endpoint https://issues.redhat.com/browse/ISPN-13834 for all event types, but the Operator is actually still using the `container/config?action=listen` endpoint. I think we need to update the docs to reflect the reality, as we can't change the current behaviour as it will break backwards compatibility.